src/rpc/rawtransaction.cpp

```diff
@@ -349,14 +355,25 @@ UniValue createrawtransaction(const JSONRPCRequest& request)
             + HelpExampleCli("createrawtransaction", "\"[{\\\"txid\\\":\\\"myid\\\",\\\"vout\\\":0}]\" \"{\\\"data\\\":\\\"00010203\\\"}\"")
             + HelpExampleRpc("createrawtransaction", "\"[{\\\"txid\\\":\\\"myid\\\",\\\"vout\\\":0}]\", \"{\\\"address\\\":0.01}\"")
             + HelpExampleRpc("createrawtransaction", "\"[{\\\"txid\\\":\\\"myid\\\",\\\"vout\\\":0}]\", \"{\\\"data\\\":\\\"00010203\\\"}\"")
+            // clang-format on
         );
+    }
 
-    RPCTypeCheck(request.params, {UniValue::VARR, UniValue::VOBJ, UniValue::VNUM}, true);
+    RPCTypeCheck(request.params,
+        {
+            UniValue::VARR,
+            UniValueType(), // ARR or OBJ, checked later
+            UniValue::VNUM
+        },
+        true);
     if (request.params[0].isNull() || request.params[1].isNull())
         throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, arguments 1 and 2 must be non-null");
 
     UniValue inputs = request.params[0].get_array();
-    UniValue sendTo = request.params[1].get_obj();
+    const bool outputs_is_array = request.params[1].isArray();
+    UniValue outputs = outputs_is_array ?
+                           request.params[1].get_array() :
+                           request.params[1].get_obj();
```

Comment:
This will raise `JSON value is not an object as expected` which doesn't match with the new signature. I suggest to switch to `isObject()` above so that `get_array` throws. In that case, it is not necessary to call`get_obj()`:
```cpp
... = outputs_is_object ? request.params[1] : request.params[1].get_array();
```

Reply:
Can you provide code to reproduce that throw, please. When I tested it a day ago in the gui, it didn't throw.

Reply:
@promag Heh, now I see what you mean. Makes sense.

Reply:
Cool, I forgot to reply! :)